src/Bicep.LangServer/Handlers/BicepGenerateParamsCommandHandler.cs

```diff
@@ -0,0 +1,146 @@
+
+            // If the template exists and contains bicep generator metadata, we can go ahead and replace the file.
+            // If not, we'll fail the generate params.
+            if (File.Exists(compiledFilePath) && !TemplateContainsSchemaField(File.ReadAllText(compiledFilePath)))
+            {
+                return "Generating parameters file failed. The file \"" + compiledFile + "\" already exists and was not generated by Bicep. If overwriting the file is intended, delete it manually and retry the Generate Params command.";
+            }
+
+            var fileUri = documentUri.ToUri();
+            RootConfiguration? configuration = null;
+
+            try
+            {
+                configuration = this.configurationManager.GetConfiguration(fileUri);
+            }
+            catch (ConfigurationException exception)
+            {
+                // Fail the generate params if there's configuration errors.
+                return exception.Message;
+            }
+
+            CompilationContext? context = compilationManager.GetCompilation(fileUri);
+            Compilation compilation;
+
+            if (context is null)
+            {
+                SourceFileGrouping sourceFileGrouping = SourceFileGroupingBuilder.Build(this.fileResolver, this.moduleDispatcher, new Workspace(), fileUri, configuration);
+                compilation = new Compilation(features, namespaceProvider, sourceFileGrouping, configuration, new LinterAnalyzer(configuration));
+            }
+            else
+            {
+                compilation = context.Compilation;
+            }
+
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == fileUri);
+
+            if (diagnosticsByFile.Value.Any(x => x.Level == DiagnosticLevel.Error))
+            {
+                return "Generating parameters file failed. Please fix below errors:\n" + DiagnosticsHelper.GetDiagnosticsMessage(diagnosticsByFile);
+            }
+
+            using var fileStream = new FileStream(compiledFilePath, FileMode.Create, FileAccess.ReadWrite);
+            var emitter = new TemplateEmitter(compilation.GetEntrypointSemanticModel(), emitterSettings);
+            EmitResult result = emitter.EmitParametersFile(fileStream);
+
+            return "Generating parameters file succeeded. Processed file " + compiledFile;
+        }
+
+        // Returns true if the template contains $schema field, false otherwise
+        public bool TemplateContainsSchemaField(string template)
+        {
+            try
+            {
+                if (!string.IsNullOrEmpty(template))
+                {
+                    JToken jtoken = template.FromJson<JToken>();
+                    var schema = jtoken.SelectToken("$schema")?.ToObject<JObject>();
+
+                    if (schema != null)
```

Comment:
Many JSON files contain schemas, you need to check that the schema contains the string "deploymentParameters" (non-case-sensitive).